docs/cpp.rst

```diff
@@ -0,0 +1,22 @@
+C++ domain customization
+========================
+
+.. confval:: cpp_strip_namespaces_from_signatures
+
+   :python:`list[str]` specifying top-level namespaces to strip from signatures.
+   This does not apply to the name of the symbol being defined by the signature,
+   only to parameter types, return types, default value expressions, etc.
+
+   For example, with the following in :file:`conf.py`:
+
+   .. code-block:: python
+
+      cpp_strip_namespaces_from_signatures = ["my_ns1", "my_ns2"]
+
+   .. rst-example:: C++ function definition with stripped namespaces
+
+      .. cpp:type:: my_ns1::A
+
+      .. cpp:type:: my_ns2::B
+
+      .. cpp:function:: void my_ns1::MyFunction(my_ns1::A x, my_ns2::B y, my_ns3::C);
```

Comment:
Given the `cpp_strip_namespaces_from_signatures` config value's setting, we could also document a `.. cpp:type:: my_ns3::C` and the namespace won't get stripped in this function's 3<sup>rd</sup> parameter, correct?